src/components/Firealerts/Form/FirealertsForm.tsx

```diff
@@ -0,0 +1,142 @@
+            method: "POST",
+            body: JSON.stringify(payload)
+        });
+        console.log(res); // TODO(gburroughs): Add notification of success/error
+    }
+
+    const getTriggerName = (trigger: FirealertsTrigger): string => {
+        const tokens = trigger.triggerName.split("-");
+        return tokens[tokens.length - 2];
+    }
+
+    if (!triggers || triggers.length === 0) {
+        return (
+            <GridCell span={12}>
+                <ZeroState />
+            </GridCell>
+        );
+    }
+    return (
+        <>
+            <GridCell span={12}>
+                {!triggers || triggers.length === 0 ? <ZeroState /> :
+                    <GridRow>
+                        <GridCell span={12}>
+                            <Typography use="headline6">Send Test {alertConfiguration[selectedAlert].name} Alert</Typography>
+                        </GridCell>
+                        <GridCell span={12}>
+                            <Callout>For more information on what the fields mean see <a href={alertConfiguration[selectedAlert].link}>here</a></Callout>
+                        </GridCell>
+                        <GridCell span={12}>
+                            <Card className="container">
+                                <Select outlined label="Select Alert Type" value={selectedAlert} options={alertsList} onChange={(e: ChangeEvent<HTMLSelectElement>) => setSelectedAlert(e.target.value as FirealertsType)}></Select>
+                                <Typography use="body2" theme="textSecondaryOnBackground"> Triggers: {implementedAlerts[selectedAlert]?.map(getTriggerName).reduce((p, c) => p + ", " + c)}</Typography>
+                                <form className="eventForm">
+                                    {createJSONForm(alertData, 1, "", updateData)}
+                                </form>
+                                <GridRow>
+                                    <GridCell span={2}>
+                                        <Button label="Send Alert" unelevated icon="send" onClick={sendAlert} />
+                                    </GridCell>
+                                </GridRow>
+                                <FirealertsLog />
+                            </Card>
+                        </GridCell>
+                    </GridRow>
+                }
+            </GridCell>
+        </>
+    );
+}
+
+const updateFromStringPath = (obj: any, path: string, value: any) => {
+    const keys = path.split(".");
+    const last = keys.pop()!;
+    let curr = obj;
+    for (const key of keys) {
+        curr = curr[key];
+    }
+    curr[last] = value;
+}
+
+
+const isObject = (x: any) => typeof x === 'object' && !Array.isArray(x) && x !== null
+
+const createIndent = (indentLevel: number) =>
+    new Array(indentLevel).fill(1).map(() => <span className="tab"></span>);
+
+const createJSONForm = (alertData: any, indentLevel: number, parent: string, update: any) => {
```

Comment:
Consider using a component instead of just a helper function

Reply:
I think this code is gonna get reworked a bit to support Array fields as well. I'll convert it over to a component when I make those changes.